src/controller/python/OpCredsBinding.cpp

```diff
@@ -343,8 +408,18 @@ PyChipError pychip_OpCreds_AllocateController(OpCredsContext * context, chip::Co
     SetDeviceAttestationVerifier(GetDefaultDACVerifier(testingRootStore));
 
     chip::Crypto::P256Keypair ephemeralKey;
-    CHIP_ERROR err = ephemeralKey.Initialize(chip::Crypto::ECPKeyTarget::ECDSA);
-    VerifyOrReturnError(err == CHIP_NO_ERROR, ToPyChipError(err));
+    chip::Crypto::P256Keypair * controllerKeyPair;
+
+    if (operationalKey == nullptr)
+    {
+        err = ephemeralKey.Initialize(chip::Crypto::ECPKeyTarget::ECDSA);
+        VerifyOrReturnError(err == CHIP_NO_ERROR, ToPyChipError(err));
+        controllerKeyPair = &ephemeralKey;
```

Comment:
The "ephemeralKey" here will actually go out of scope. We cannot have `ephemeralKey` be here by value. It should be a member at the very least. We should avoid following the flow where the SDK makes a "copy" of the keypair using Serialize/Deserialize

Reply:
By using `hasExternallyOwnedOperationalKeypair`, the SDK won't copy the keypair, and we (Python) are responsible for recovering the credentials.
(Currently we don't need this if we are only doing commissioning)